addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -241,14 +245,17 @@ def action_done(self):
                     picking.batch_id.id,
                     picking.batch_id.name))
 
+        # Run sanity_check as a batch and ignore the one in button_validate() since it is done here.
+        pickings._sanity_check(separate_pickings=False)
+        # Skip sanity_check in pickings button_validate() & remove 'waiting' pickings from the batch
+        context = {'skip_sanity_check': True, 'pickings_to_detach': empty_waiting_pickings.ids}
```

Comment:
I'm conflicted if it's better to pass `separate_pickings=False`  in a context so you don't need to pass the `skip_sanity_check` context and do the _sanity_check separately :thinking: but it probably doesn't matter that much even if there will never be another case where we skip the sanity check.